### PR TITLE
Add bpfman OpenShift container to build bpfman

### PR DESCRIPTION
**Pull request by msherif1234**
Openshift container to build bpfman and used by konflux pipeline

### CODE REVIEW

File: Containerfile.bpfman.openshift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ARG TARGETOS=linux
+ARG TARGETPLATFORM=linux/amd64
+
+RUN echo "TARGETOS=${TARGETOS}  TARGETARCH=${TARGETARCH}  BUILDPLATFORM=${BUILDPLATFORM}  TARGETPLATFORM=${TARGETPLATFORM}"
+
+WORKDIR /usr/src/bpfman
+COPY ./ /usr/src/bpfman
+
+RUN ${DNF_CMD} update -y \
+    && ${DNF_CMD} install -y rust cargo openssl-devel \
```
Comment (anfredette) [inline Containerfile.bpfman.openshift:20]: Do you know what version of cargo we're getting?

**Comment by anfredette** [comment]
In general, this looks good to me, I'm just wondering what versions of rust and cargo we're getting and whether there could be any compatibility issues.

File: Containerfile.bpfman.openshift (hunk 1 above)
Comment (frobware) [reply Containerfile.bpfman.openshift:20]:
```console
% podman run -it --rm registry.access.redhat.com/ubi9/ubi-minimal:9.4-1194 /bin/bash
[root@c964ec992e9a /]# microdnf update && microdnf repoquery cargo

(microdnf:8): librhsm-WARNING **: 09:25:15.384: Found 0 entitlement certificates

(microdnf:8): librhsm-WARNING **: 09:25:15.386: Found 0 entitlement certificates
Downloading metadata...
Downloading metadata...
Downloading metadata...
Package                                                                                                                                        Repository                                              Size
Upgrading:
 curl-minimal-7.76.1-29.el9_4.1.x86_64                                                                                                         ubi-9-baseos-rpms                                   131.8 kB
  replacing curl-minimal-7.76.1-29.el9_4.x86_64
 libcurl-minimal-7.76.1-29.el9_4.1.x86_64                                                                                                      ubi-9-baseos-rpms                                   233.7 kB
   replacing libcurl-minimal-7.76.1-29.el9_4.x86_64
Transaction Summary:
 Installing:        0 packages
 Reinstalling:      0 packages
 Upgrading:         2 packages
 Obsoleting:        0 packages
 Removing:          0 packages
 Downgrading:       0 packages
Is this ok [y/N]: y
Downloading packages...
Running transaction test...
Updating: libcurl-minimal;7.76.1-29.el9_4.1;x86_64;ubi-9-baseos-rpms
Updating: curl-minimal;7.76.1-29.el9_4.1;x86_64;ubi-9-baseos-rpms
Cleanup: curl-minimal;7.76.1-29.el9_4;x86_64;installed
Cleanup: libcurl-minimal;7.76.1-29.el9_4;x86_64;installed
Complete.

(microdnf:66): librhsm-WARNING **: 09:25:18.082: Found 0 entitlement certificates

(microdnf:66): librhsm-WARNING **: 09:25:18.084: Found 0 entitlement certificates

cargo-1.75.0-1.el9.x86_64
```

**Comment by msherif1234** [comment]
> In general, this looks good to me, I'm just wondering what versions of rust and cargo we're getting and whether there could be any compatibility issues.

this what we have in rhel9.4 it seems rhel9.5 will have `rust 1.79` but this isn't something we have control over

**Comment by Billy99** [comment]
Looks like currently stable has 1.80.1 and nightly is on 1.82.0. So there will probably be issues at some point.

File: Containerfile.bpfman.openshift (hunk 1 above)
Comment (danielmellado) [inline Containerfile.bpfman.openshift:20]: We're  still bundling a version of the fiat-crypto crate.
This crate contains implementations of elliptic-curve cryptography that is *NOT* approved to be shipped by Fedora/RH *in any form* (i.e. also not as source code).
We will need to patch out any references to the p434 curve *before* compressing the vendor tarball.
We can take the patch from the Fedora package for the crate (rust-fiat-crypto).

**Review by anfredette — APPROVED** [review]
I'm going to approve this to keep things moving, but we can revisit it later if issues turn up.

**Pull request merged**